BlinkIDSample/BlinkID-ComposeFragmentSample/src/main/java/com/microblink/blinkid/MainActivity.kt

```diff
@@ -0,0 +1,330 @@
+package com.microblink.blinkid
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.compose.BackHandler
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.viewinterop.AndroidViewBinding
+import com.microblink.blinkid.databinding.RecognizerRunnerLayoutBinding
+import com.microblink.blinkid.entities.recognizers.RecognizerBundle
+import com.microblink.blinkid.entities.recognizers.blinkid.generic.BlinkIdMultiSideRecognizer
+import com.microblink.blinkid.entities.recognizers.blinkid.generic.BlinkIdSingleSideRecognizer
+import com.microblink.blinkid.fragment.RecognizerRunnerFragment
+import com.microblink.blinkid.fragment.overlay.ScanningOverlay
+import com.microblink.blinkid.fragment.overlay.blinkid.BlinkIdOverlayController
+import com.microblink.blinkid.fragment.overlay.blinkid.BlinkIdOverlaySettings
+import com.microblink.blinkid.fragment.overlay.blinkid.BlinkIdOverlayView
+import com.microblink.blinkid.fragment.overlay.blinkid.reticleui.ReticleOverlayStrings
+import com.microblink.blinkid.fragment.overlay.blinkid.reticleui.ReticleOverlayView
+import com.microblink.blinkid.recognition.RecognitionSuccessType
+import com.microblink.blinkid.uisettings.BlinkIdUISettings
+import com.microblink.blinkid.view.recognition.ScanResultListener
+
+class MainActivity : AppCompatActivity(), RecognizerRunnerFragment.ScanningOverlayBinder {
```

Comment:
Opcenito je [bolje naslijediti ComponentActivity u Compose appovima](https://stackoverflow.com/a/67891505) ako ne koristimo AndroidViewove (koji trebaju appbar). U onom drugom sampleu je i okej jer si trebao brzi tooplbar, pa ajde. Al posto je ovo ipak Compose based app, mozes koristiti Scaffold koji ima built in `topBar` i mozes ga prikazivati ili ne ovisno o ovom stanju kojeg drzis.
Ovak bi maknuo ove LaunchedEffecte i supportActionBar.show() i hide()

Reply:
Je, ali imam problem s onConfigurationChanged onda, jer ne mogu zvat supportFragmentManager(Koji se nalazi u FragmentActivity i AppCompatActivity klasi). Taj dio mi je koristan jer jedino tako mogu projverit je li već fragment inflatean ili ne. Prije nego što sam stavio onaj block onConfigurationChanged s onim svim provjerama, jako puno raznih crasheva sam imao, race conditioni, null pointeri itd. Ako postoji neka alternativa tome, mogu, ali ovo je ipak Fragment implementacija a ne tipicni Compose app.